cmake/CommonModularInterface.cmake

```diff
@@ -32,9 +45,15 @@ FOREACH(file ${MODULAR_FILES})
 	)
 ENDFOREACH()
 
-ADD_CUSTOM_TARGET(${MODULAR_NAME}_modular_src
-	DEPENDS shogun ${modular_files}
-	COMMENT "copying SWIG files")
+IF(JINJA2_FOUND)
+	ADD_CUSTOM_TARGET(${MODULAR_NAME}_modular_src
+		DEPENDS shogun ${modular_files} shogun-base.i
```

Comment:
you could just append `shogun-base.i` to the `${modular_files}`, and call `ADD_CUSTOM_TARGET` once.